File: RLE.CSharp.Tests/DecoderTests.cs
using Xunit;

namespace RLE.CSharp.Tests
{
    public class DecoderTests
    {
        [Fact]
        public void Can_decode_single_tuple() => 
            Assert.Equal("a", Decoder.Process("1a"));

        [Fact]
        public void Can_decode_single_tuple_multiple_chars() => 
            Assert.Equal("aaa", Decoder.Process("3a"));

        [Fact]
        public void Can_decode_multiple_tuple_single_chars() => 
            Assert.Equal("abc", Decoder.Process("1a1b1c"));

        [Fact]
        public void Can_decode_multiple_tuple_multiple_chars() => 
            Assert.Equal("aabbbc", Decoder.Process("2a3b1c"));

        [Fact]
        public void Can_decode_multiple_tuple_multiple_chars_with_repeats() => 
            Assert.Equal("aabbbcaaaaxbbbbbcc", Decoder.Process("2a3b1c4a1x5b2c"));
    }
}

File: src/DecoderTests.java
import org.junit.jupiter.api.Test;

import java.security.InvalidParameterException;

import static org.junit.jupiter.api.Assertions.*;

class DecoderTests {
    @Test
    public void Will_throw_if_input_not_alphanumeric() {
        assertThrows(InvalidParameterException.class, () -> Decoder.Process("!"));
    }

    @Test
    public void Will_throw_if_input_contains_non_alphanumeric() {
        assertThrows(InvalidParameterException.class, () -> Decoder.Process("1a1b!"));
    }

    @Test
    public void Can_decode_single_tuple() {
        assertEquals("a", Decoder.Process("1a"));
    }

    @Test
    public void Can_decode_single_tuple_multiple_chars() {
        assertEquals("aaa", Decoder.Process("3a"));
    }

    @Test
    public void Can_decode_multiple_tuple_single_chars() {
        assertEquals("abc", Decoder.Process("1a1b1c"));
    }

    @Test
    public void Can_decode_multiple_tuple_multiple_chars() {
        assertEquals("aabbbc", Decoder.Process("2a3b1c"));
    }

    @Test
    public void Can_decode_multiple_tuple_multiple_chars_with_repeats() {
        assertEquals("aabbbcaaaaxbbbbbcc", Decoder.Process("2a3b1c4a1x5b2c"));
    }
}